### ISSUE TITLE
Loot table kept crashing after adding "set_lore" modifier

### ISSUE THREAD

**Issue by akastrmix** [misode/misode.github.io#648]
This is all that I remember that happened. I wrote a bunch of stuff in the loot table generator, and right after I added "set_lore", the page crashed and showed this image
![image](https://github.com/user-attachments/assets/655e089e-1683-4b25-9612-569c81bb98d5)
And since the webpage caches all the content, every time I open the loot table page, it will instantly crash.
This has caused several troubles to my work progress, please consider fixing it as quickly as possible.

**Comment by DlljsCodes** [comment]
Also experiencing this crash. Here's my full crash report:
## Crash report
 * Page url: `https://misode.github.io/loot-table/`
 * Generator ID: `loot_table`
 * Current version: `1.21.2`
 * Latest version: `1.21.4-rc2`

### Stack trace
```
TypeError: Something went wrong rendering the generator: Cannot read properties of undefined (reading 'flatMap')
    at flatMap (src/app/components/previews/LootTable.ts:474:31)
    at item (src/app/components/previews/LootTable.ts:279:44)
    at compositeFunction (src/app/components/previews/LootTable.ts:267:2)
    at consumer (src/app/components/previews/LootTable.ts:268:2)
    at entryConsumer (src/app/components/previews/LootTable.ts:238:3)
    at entries (src/app/components/previews/LootTable.ts:169:15)
    at pool (src/app/components/previews/LootTable.ts:130:15)
    at generateTable (src/app/components/previews/LootTable.ts:48:1)
    at generateLootTable (src/app/components/previews/LootTablePreview.tsx:53:9)
    at n (../../node_modules/preact/hooks/dist/hooks.module.js:1:902)
```

### Generator JSON
<details>
<pre>
{
  "pools": [
    {
      "rolls": 1,
      "entries": [
        {
          "type": "minecraft:item",
          "name": "minecraft:acacia_leaves"
        }
      ],
      "functions": [
        {
          "function": "minecraft:set_lore"
        }
      ]
    }
  ]
}
</pre>
</details>

**Comment by misode** [comment]
Fixed in a8aaec69e2f91378415d3d6fb3ac9b4d061b3a27 - sorry for the long wait!
